packages/clay-badge/src/ClayBadge.soy

```diff
@@ -0,0 +1,122 @@
+{namespace ClayBadge}
+
+/**
+ * This renders a badge.
+ */
+{template .render}
+	{@param? icon: [
+		alignment: string,
+		monospaced: bool,
+		spritemap: string,
+		symbol: string
+	]}
+	{@param? id: string}
+	{@param? label: html|string}
+	{@param? href: string}
+	{@param? shape: string}
+	{@param? size: string}
+	{@param? style: string}
+	{@param? linkWrapper: bool}
+	{@param? onClickIcon: any}
+	{@param? onClickLink: any}
+
+	{let $attributes kind="attributes"}
+		class="badge
+			{sp}badge-{$style ?: 'default'}
+
+			{if $shape == 'pill'}
+				{sp}badge-pill
+			{/if}
+
+			{if $size}
+				{sp}badge-{$size}
+			{/if}
+		"
+
+		{if $id}
+			id="{$id}"
+		{/if}
+
+		{if $href}
+			href="{$href}"
+		{/if}
+	{/let}
+
+	{if $href and $linkWrapper}
```

Comment:
From these checks, looks like I cannot have the whole badge wrapped around an anchor while at the same time have an icon displayed. Maybe it's an option we want people to be able to pick. What do you think?